homeassistant/components/google_assistant/smart_home.py

```diff
@@ -105,14 +108,38 @@ def entity_to_device(entity: Entity):
             if m in CLIMATE_SUPPORTED_MODES)
         device['attributes'] = {
             'availableThermostatModes': modes,
-            'thermostatTemperatureUnit': 'C',
+            'thermostatTemperatureUnit':
+            'F' if units.temperature_unit == TEMP_FAHRENHEIT else 'C',
         }
 
     return device
 
 
-def query_device(entity: Entity) -> dict:
+def query_device(entity: Entity, units: UnitSystem) -> dict:
     """Take an entity and return a properly formatted device object."""
+    def celsius(deg: Optional[float]) -> Optional[float]:
+        """Convert a float to Celsius and rounds to one decimal place."""
+        if deg is None:
+            return None
+        return round(METRIC_SYSTEM.temperature(deg, units.temperature_unit), 1)
+    if entity.domain == climate.DOMAIN:
+        mode = entity.attributes.get(climate.ATTR_OPERATION_MODE)
+        response = {
+            'thermostatMode':
+            mode if mode in CLIMATE_SUPPORTED_MODES else 'on',
```

Comment:
Not a fan of inline if statements. Hurts readability.

Reply:
Replaced with normal if statement.